app/src/main/java/org/mozilla/focus/tabs/MergeTabsMiddleware.kt

```diff
@@ -58,9 +58,9 @@ class MergeTabsMiddleware(
         // engine session to this tab.
         next(TabListAction.AddTabAction(mergedTab, select = true))
 
-        // Then we can remove the previous tab. We first unlink the engine session to prevent the
-        // middleware from closing the engine session, which is now linked to the new tab.
-        context.dispatch(EngineAction.UnlinkEngineSessionAction(currentTab.id))
```

Comment:
I would keep a comment here because this is definitely not self-explanatory.

If the new tab does not have an engine session we will re-use the engine session of the current tab so we unlink it first to prevent the engine middleware from closing it.

If the new tab has an engine session (e.g. when it was opened by a web extension) we will use the new engine session, close the current tab incl. its engine session.